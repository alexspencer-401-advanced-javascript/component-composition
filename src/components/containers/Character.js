import React, { Component } from 'react';
import DeckOfCards from '../Decks/DeckOfCards';
import { getCharacters } from '../services/arnoldApi';

export default class Character extends Component {
  state = {
    characters: []
  }

  componentDidMount(){
    return getCharacters()
      .then((fetchedChars) => {
        this.setState({ 
          characters: fetchedChars 
        });
      });
  }

  render(){
    return (
      <DeckOfCards items={this.state.characters}/>
    );
  }
}
