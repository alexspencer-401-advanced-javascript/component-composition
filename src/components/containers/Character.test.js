import React from 'react';
import { shallow } from 'enzyme';
import Character from './Character';


jest.mock('../services/arnoldApi.js', () => ({
  getCharacters() {
    return Promise.resolve([
      {
        '_id': '5da237699734fdcb7bef8f56',
        'name': 'Miles Shortman',
        'image': 'https://vignette.wikia.nocookie.net/heyarnold/images/2/2e/Miles.jpg/revision/latest/scale-to-width-down/243?cb=20101115140639'
      }]);
  }
}));

describe('image component', () => {
  it('makes a successful api call and returns characters', () => {
    const wrapper = shallow(<Character />);
    expect(wrapper).toMatchSnapshot();
  });
});
