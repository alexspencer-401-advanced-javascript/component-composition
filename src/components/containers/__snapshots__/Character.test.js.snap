// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`image component makes a successful api call and returns characters 1`] = `
<DeckOfCards
  props={Array []}
/>
`;
